extension.js

```diff
@@ -273,27 +272,45 @@ async function build(config, componentPath, server) {
 		}
 
 		// Build
-		const [command, ...args] = shellQuote.parse(config.buildCommand);
+		try {
+			logger.info(`Building Next.js application at ${componentPath}`);
+
+			const timerStart = performance.now();
+
+			const stdout = [];
+			const stderr = [];
 
-		const timerStart = performance.now();
+			const buildProcess = spawn(config.buildCommand, [], {
+				shell: true,
+				cwd: componentPath,
+				stdio: ['ignore', 'pipe', 'pipe'],
+			});
 
-		const { stdout, stderr, status, error } = spawnSync(command, args, {
-			cwd: componentPath,
-			encoding: 'utf-8',
-		});
+			buildProcess.stdout.on('data', c => stdout.push(c));
+			buildProcess.stderr.on('data', c => stderr.push(c));
+
+			const [code, signal] = await once(buildProcess, 'exit');
```

Comment:
Consider using `close` instead of `exit`. The `close` event is emitted once stdio has been flushed.

```suggestion
			const [code, signal] = await once(buildProcess, 'close');
```

BTW, this API is really nice. I need to start using `once()`!